stable/database/templates/_helpers.tpl

```diff
@@ -616,3 +624,23 @@ rendered, false otherwise.
 {{- end -}}
 {{ $ret }}
 {{- end -}}
+
+{{/*
+Truncates a string to the max length of characters. If the original string
+length exceeds the supplied max length, a 7 characters SHA256 hash is
+calculated and appended to the truncated string to make it unique.
+*/}}
+{{- define "truncWithHash" -}}
+{{- $s := index . 0 -}}
+{{- $max := index . 1 -}}
+{{- if not (typeIs "string" $s) -}}
+{{- fail (printf "truncWithHash template failed: not supported type '%s'" (typeOf $s)) -}}
+{{- end -}}
+{{- if gt (len $s) $max -}}
```

Comment:
Is there a minimum max value we should enforce? Obviously, you can't pass in max less than or equal to 8 since we reserve space for the hash suffix.

Reply:
Good point! I will add validation for that.